### ISSUE TITLE
Trailing whitespace causes parser error

### ISSUE THREAD

**Issue by jyc** [xinz/sax_map#7]
Hello! Thanks for making this library.

I ran into an error while trying to parse a RSS feed. I've reduced the test case to this:

```patch
diff --git a/test/sax_map_test.exs b/test/sax_map_test.exs
index 5aa1245..ace7c7e 100644
--- a/test/sax_map_test.exs
+++ b/test/sax_map_test.exs
@@ -392,4 +393,15 @@ defmodule SAXMapTest do
     assert map["xml"]["Content"] == "\n  ."
   end
 
+  test "parse trailing whitespace" do
+    # Note the single trailing space character after <a></a>.
+    xml = """
+    <xml>
+    <a></a> 
+    <b></b>
+    </xml>
+    """
+
+    {:ok, _map} = SAXMap.from_string(xml)
+  end
 end
```

... which yields the following error when `mix test test/sax_map_test.exs:396` is run:

```
  1) test parse trailing whitespace (SAXMapTest)
     test/sax_map_test.exs:396
     ** (FunctionClauseError) no function clause matching in SAXMap.Handler.list_to_map/2

     The following arguments were given to SAXMap.Handler.list_to_map/2:

         # 1
         " \n"

         # 2
         %{"b" => nil}

     Attempted function clauses (showing 3 out of 3):

         defp list_to_map([], prepared)
         defp list_to_map([item | rest], prepared) when is_map(item)
         defp list_to_map([{key, value} | rest], prepared)

     code: {:ok, _map} = SAXMap.from_string(xml)
     stacktrace:
       (sax_map 1.1.0) lib/handler.ex:227: SAXMap.Handler.list_to_map/2
       (sax_map 1.1.0) lib/handler.ex:198: SAXMap.Handler.handle_event/3
       (saxy 1.5.0) lib/saxy/emitter.ex:16: Saxy.Emitter.emit/3
       (saxy 1.5.0) lib/saxy/emitter.ex:7: Saxy.Emitter.emit/4
       (saxy 1.5.0) lib/saxy/parser/builder.ex:1230: Saxy.Parser.Binary.element_misc/5
       (saxy 1.5.0) lib/saxy.ex:194: Saxy.parse_string/4
       test/sax_map_test.exs:405: (test)
```

I tried looking into the code but wasn't sure about the big picture. From what I can tell:

- `handle_event(:end_document, ...)` ends up with the state containing an improper list, `[{"b", nil} | " \n"]`
- `format_key_value_pairs` gets this improper list, then passes it to
- `list_to_map`, which recurses on the tail of the list, which fails because no `list_to_map` case handles a non-list item

If you're able to write up some comments on how the code is supposed to work I'd be happy to dive into it more!

**Comment by xinz** [comment]
Hi @jyc , thanks very much for your feedback, there is a fixing in the `master` branch, could you please try to use that and double check it is working for you?

**Comment by xinz** [comment]
I submitted some change updates, please use the latest commit [4348c2d](https://github.com/xinz/sax_map/commit/4348c2d978b41b3577f10dba5855618b357cac4e) for the testing and review, thanks.

**Comment by jyc** [comment]
Sorry for the delayed reply, thank you for the fix!!

**Comment by xinz** [comment]
No rush, thanks for your help, now the fixing is released in [1.2.0](https://hex.pm/packages/sax_map/1.2.0)
